app/server.js

```diff
@@ -0,0 +1,204 @@
+      return done(null, profile);
+    });
+  }
+));
+
+passport.use(new LocalStrategy( (username, password, done)=> {
+  printMessage(username, password);
+
+
+  done();
+  // query database to get user details
+  
+}))
+
+
+// configure Express
+var app = express();
+app.set('views', __dirname + '/views');
+app.set('view engine', 'ejs');
+app.use(partials());
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
+app.use(methodOverride());
+
+app.use(session({ secret: 'adam alinoe', resave: false, saveUninitialized: false }));
+
+app.use(passport.initialize());
+app.use(passport.session());
+app.use(express.static(__dirname + '/public'));
+
+app.get('/', (req, res) => res.render('index', { user: req.user }));
+
+app.get('/login', (req, res) => {
+  res.render('login', { user: req.user });
+});
+
+app.post('/login-user', (req, res) => {
+  passport.authenticate('local', {successRedirect: '/account', failureRedirect: '/login', failureFlash: true })
+
+})
+
+app.get('/auth/github', passport.authenticate('github', { scope: ['user:email'] }), (req, res) => { /*tutaj nigdy nie bedziemy, idz do callback*/ });
+
+app.get('/auth/github/callback', passport.authenticate('github', { failureRedirect: '/login' }), (req, res) => {
+  res.redirect('/account');
+});
+
+app.get('/account', ensureAuthenticated, co.wrap(function* (req, res) {
+  try { 
+    const user = req.user._json;
+    printMessage('user', user);
+
+    let queryString = `UPDATE user_details set login='${user.login}', github_user_id=${user.id}, html_url='${user.html_url}', repos_url='${user.repos_url}', token='${user.token}' where EXISTS (SELECT * FROM user_details WHERE user_details.github_user_id = ${user.id});
+        INSERT INTO user_details (login, github_user_id, html_url, repos_url, token)  
+        SELECT '${user.login}', ${user.id}, '${user.html_url}', '${user.repos_url}', '${user.token}'
+        WHERE NOT EXISTS (SELECT 1 FROM user_details WHERE user_details.github_user_id = ${user.id});
+
+        SELECT user_details.id, roles.name as roleName FROM user_details, roles
+        WHERE user_details.github_user_id = ${user.id}
+        AND user_details.role_id = roles.id;`;
+
+    user.userDetails = yield dbService.queryDatabase(queryString);
+    req.session.user = user;
+    printMessage('user after log in', user);
+    printMessage('user.userDetails', user.userDetails.rolename)
+    
+    res.render('account', { user, title: 'Moje konto', roleName: user.userDetails.rolename });
+
+  } catch (e) {
+    printMessage('got error', e);
+  }
+}));
+
+
+app.get('/repositories', ensureAuthenticated, (req, res) => {
+
+  const user = req.session.user;
+
+  const storeRepositories = (body) => {
+    let repositories = JSON.parse(body);
+      // `UPDATE repositories set git_project_id='${item.id}', project_name='${item.name}', full_project_name='${item.full_name}', html_url='${item.html_url}', description='${item.description}', api_url='${item.url}'
```

Comment:
byl tam wczesniej update or insert ale nie dzialalo to razem jak tzreba. Zamienilem to na ON CONFLICT
```
    coForEach(repositories, function* (item, index) {
      let queryString = `INSERT INTO repositories (user_id, git_project_id, project_name, full_project_name, html_url, description, api_url)
        SELECT '${user.id}', '${item.id}','${item.name}', '${item.full_name}', '${item.html_url}', '${item.description}', '${item.url}'
        ON CONFLICT (git_project_id) DO UPDATE
        SET project_name='${item.name}', full_project_name='${item.full_name}', html_url='${item.html_url}', description='${item.description}', api_url='${item.url}';`

      yield dbService.queryDatabase(queryString);

    })
```